backend/src/test/java/ch/puzzle/okr/service/validation/TeamValidationServiceTest.java

```diff
@@ -0,0 +1,168 @@
+package ch.puzzle.okr.service.validation;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import ch.puzzle.okr.models.Team;
+import ch.puzzle.okr.service.persistance.TeamPersistenceService;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.InjectMocks;
+import org.mockito.Mockito;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+class TeamValidationServiceTest {
+    @MockBean
+    TeamPersistenceService teamPersistenceService = Mockito.mock(TeamPersistenceService.class);
+
+    Team team1;
+    Team team2;
+    Team teamWithIdNull;
+
+    @BeforeEach
+    void setUp() {
+        team1 = Team.Builder.builder().withId(1L).withName("Team 1").build();
+        team2 = Team.Builder.builder().withId(2L).withName("Team 2").build();
+        teamWithIdNull = Team.Builder.builder().withId(null).withName("Team null").build();
+
+        when(teamPersistenceService.findById(1L)).thenReturn(team1);
+        when(teamPersistenceService.getModelName()).thenReturn("Team");
+        doThrow(new ResponseStatusException(HttpStatus.NOT_FOUND,
+                String.format("%s with id %s not found", teamPersistenceService.getModelName(), 2L)))
+                        .when(teamPersistenceService).findById(2L);
+    }
+
+    @Spy
```

Comment:
Why Spy?

Reply:
bro i genuinely don't know

Reply:
nvm this is like this so i can oberserve and verify function calls